### PR TITLE
PM-11479 Expired link UI

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-11479

## 📔 Objective
- Build UI layer for the expired registration email link screen.

## 📸 Screenshots
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    @Test
+    fun `NavigateBack event invokes onNavigateBack`() {
+        mutableEventFlow.tryEmit(ExpiredRegistrationLinkEvent.NavigateBack)
+        assert(onNavigateBackCalled)
+    }
+
+    @Test
+    fun `NavigateToLogin event invokes onNavigateToLogin`() {
+        mutableEventFlow.tryEmit(ExpiredRegistrationLinkEvent.NavigateToLogin)
+        assert(onNavigateToLoginCalled)
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkScreenTest.kt:73]: Can we swap these out for `assertTrue`

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/auth/AuthNavigation.kt [hunk 2]
```diff
@@ -189,6 +190,23 @@ fun NavGraphBuilder.authGraph(
...
+                navController.navigateToStartRegistration(
+                    navOptions = navOptions {
+                        popUpTo(LANDING_ROUTE)
+                    },
+                )
+            },
+            onNavigateToLogin = {
+                navController.navigateToLanding(
+                    navOptions = navOptions {
+                        popUpTo(LANDING_ROUTE)
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/auth/AuthNavigation.kt:205]: This doesn't put a second LandingScreen on the stack, right?

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/auth/AuthNavigation.kt:205]: no but you beat me to pushing a change to address some funkiness I messed up with the root nav screen and rootNavOptions

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkScreenTest.kt (hunk 1 above)
Comment (dseverns-livefront) [reply app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkScreenTest.kt:73]: d'oh, gr8 catch

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt [hunk 3]
```diff
@@ -156,6 +158,11 @@ fun RootNavScreen(
                 )
             }
 
+            RootNavState.ExpiredRegistrationLink -> {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt:161]: What is the intended way to prompt this screen?

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt:161]: follow up PR, but quick hits version, we get CompleteReg intent, we check the token via endpoint if valid we do what we do currently and set special circumstance to complete registration, else if expired set it to one that will cause root nav to here. I can post that PR, but was just waiting for this to get merged assuming I'd have some sub commits and saving rebase hell :)

Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt:161]: Gotcha, makes sense to me!

File: app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.Test
+
+class ExpiredRegistrationLinkViewModelTest : BaseViewModelTest() {
+
+    @Test
+    fun `CloseClicked sends NavigateBack event`() = runTest {
+        val viewModel = ExpiredRegistrationLinkViewModel()
+        viewModel.trySendAction(ExpiredRegistrationLinkAction.CloseClicked)
+        viewModel.eventFlow.test {
+            assertEquals(ExpiredRegistrationLinkEvent.NavigateBack, awaitItem())
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/expiredregistrationlink/ExpiredRegistrationLinkViewModelTest.kt:16]: Can we move the `trySendAction` into the test lambda

**Review by david-livefront — APPROVED** [review]

**Pull request merged**